jest-resolve imports IModuleMap, removed in v27.0.5 of jest-haste-map

<!-- Love Jest? Please consider supporting our collective: 👉  https://opencollective.com/jest/donate -->

## 🐛 Bug Report
Version 27.0.5 of `jest-haste-map` removes IModuleMap, which is used by `jest-resolve`
```
$ tsc
node_modules/jest-resolve/build/resolver.d.ts:8:15 - error TS2724: '"../../jest-haste-map/build"' has no exported member named 'IModuleMap'. Did you mean 'ModuleMap'?

8 import type { IModuleMap } from 'jest-haste-map';
                ~~~~~~~~~~

  node_modules/jest-haste-map/build/index.d.ts:180:12
    180     static ModuleMap: typeof HasteModuleMap;
                   ~~~~~~~~~
    'ModuleMap' is declared here.


Found 1 error.
```

## To Reproduce

Steps to reproduce the behavior:
Build a project with jest-resolve in the dependency tree

## Expected behavior

<!-- A clear and concise description of what you expected to happen. -->
Builds without error (as with v <27.0.5)
## Link to repl or repo (highly encouraged)

<!--
Please provide either a [repl.it demo](https://repl.it/languages/jest) or a minimal repository on GitHub.
Issues without a reproduction link are likely to stall.

See https://stackoverflow.com/help/minimal-reproducible-example for information on how to create good reproductions
-->

## envinfo

<!--
Run npx envinfo --preset jest
Paste the results here:
-->

```
  System:
    OS: macOS 10.15.7
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
  Binaries:
    Node: 12.15.0 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 6.13.4 - /usr/local/bin/npm
  npmPackages:
    jest: ^27.0.5 => 27.0.5 
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,7 @@
 
 - `[jest-environment-node]` Add `Event` and `EventTarget` to node global environment. ([#11705](https://github.com/facebook/jest/issues/11705))
 - `[jest-mock]` Fix `spyOn` to use `Object.prototype.hasOwnProperty` [#11721](https://github.com/facebook/jest/pull/11721)
+- `[jest-resolver]` Add dependency on `jest-haste-map` [#11759](https://github.com/facebook/jest/pull/11759)
 
 ### Chore & Maintenance
 

--- a/packages/jest-resolve/package.json
+++ b/packages/jest-resolve/package.json
@@ -18,6 +18,7 @@
     "chalk": "^4.0.0",
     "escalade": "^3.1.1",
     "graceful-fs": "^4.2.4",
+    "jest-haste-map": "^27.0.6",
     "jest-pnp-resolver": "^1.2.2",
     "jest-util": "^27.0.6",
     "jest-validate": "^27.0.6",
@@ -26,8 +27,7 @@
   },
   "devDependencies": {
     "@types/graceful-fs": "^4.1.3",
-    "@types/resolve": "^1.20.0",
-    "jest-haste-map": "^27.0.6"
+    "@types/resolve": "^1.20.0"
   },
   "engines": {
     "node": "^10.13.0 || ^12.13.0 || ^14.15.0 || >=15.0.0"

